kclvm/compiler_base/error/src/diagnostic/components.rs

```diff
@@ -52,3 +57,394 @@ impl Component<DiagnosticStyle> for Label {
         }
     }
 }
+
+/// `UnderLine` is a component of diagnostic to display an underline.
+///
+/// ```ignore
+/// int test = 0;
+///     ^^^^ This is an underline under variable `test`
+/// ```
+pub struct UnderLine {
+    start: usize,
+    end: usize,
+    label: String,
+    style: Option<DiagnosticStyle>,
```

Comment:
I prefer that the style is decoupled with each component itself
style take effect to all components and is not a part of any component